client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SparkUtils.java

```diff
@@ -477,22 +473,36 @@ public static boolean shouldReportShuffleFetchFailure(long taskId) {
                   taskId,
                   taskInfo.attemptNumber(),
                   ti.attemptNumber());
-              return false;
-            }
-          } else {
-            if (ti.attemptNumber() >= maxTaskFails - 1) {
-              LOG.warn(
-                  "StageId={} index={} taskId={} attemptNumber {} reach maxTaskFails {}.",
+              hasRunningAttempt = true;
+            } else if (ti.status() == "FAILED") {
```

Comment:
It needs to define the same task status scope as Spark, such as KILLED or UNKNOWN.